vassal-app/src/main/java/VASSAL/build/widget/PieceSlot.java

```diff
@@ -192,13 +192,14 @@ protected void clearExpandedPiece() {
    */
   public GamePiece getPiece() {
     if (c == null && pieceDefinition != null) {
-      final AddPiece comm =
-        (AddPiece) GameModule.getGameModule().decode(pieceDefinition);
-      if (comm == null) {
+      final Command cp = GameModule.getGameModule().decode(pieceDefinition); 
+      if ((cp == null) || cp.isNull() || !(cp instanceof AddPiece)) {
         System.err.println("Couldn't build piece " + pieceDefinition);
```

Comment:
Uh wow, we're printing the message to stderr?! That should go the Chatter, like Bad Data Errors do.